### PR TITLE
Non generic batching

### PR DESCRIPTION
**Pull request by Donatas-L**

### CODE REVIEW

File: src/client/client.ts [hunk 1]
```diff
@@ -144,22 +144,21 @@ export class TravelTimeClient {
...
+    const results: BatchResponse<R>[] = [];
 
-      // eslint-disable-next-line no-await-in-loop
-      const chunkResults = await Promise.allSettled(promises);
-      chunkResults.forEach((chunkResult) => {
-        if (chunkResult.status === 'rejected') {
-          results.push({ type: 'error', error: chunkResult.reason });
+    promises.forEach(promise => {
+        if (promise.status === 'rejected') {
+          results.push({ type: 'error', error: promise.reason });
```
Comment (danielnaumau) [inline src/client/client.ts:157]: MUTABILITY !!!!

File: src/client/client.ts [hunk 2]
```diff
@@ -144,22 +144,21 @@ export class TravelTimeClient {
     bodies: Parameters<T>[0][],
     chunkSize = 10,
   ): Promise<BatchResponse<R>[]> {
-    const results: BatchResponse<R>[] = [];
+    const slices: Parameters<T>[][] = Array
+      .from({ length: (bodies.length / chunkSize) }, (_, index) => index)
+      .map((i) => bodies.slice(i * chunkSize, (i + 1) * chunkSize));
+
+    const promises = await Promise.allSettled(slices.flatMap((requests => requests.map(request => requestFn(request)))));
```
Comment (danielnaumau) [inline src/client/client.ts:151]: I still don't get what's the difference between all and allSettled

**Comment by Donatas-L** [comment]
Do we even consider merging this as well? Or should I close it?

**Comment by danielnaumau** [comment]
> Do we even consider merging this as well? Or should I close it?

let's close it and just wait when @EivydasKoc comes back from vacation

**Pull request closed without merging**